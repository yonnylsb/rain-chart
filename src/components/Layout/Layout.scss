body {
    background: rgb(252,252,252);
    background: linear-gradient(90deg, rgba(252,252,252,1) 0%, rgba(250,248,255,1) 100%);
}

.layout {
    padding: 20px;
}