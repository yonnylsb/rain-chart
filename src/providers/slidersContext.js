import React, { useState } from 'react';
import rainchartConstants from '../common/constants';

export const SlidersContext = React.createContext();

export default function SliderContextProvider(props) {
    let [pressureGraph, setPressureGraph] = useState(rainchartConstants.pressure.min);
    let [temperatureGraph, setTemperatureGraph] = useState(rainchartConstants.temperature.min);
    let context = {
        pressureGraph,
        setPressureGraph,
        temperatureGraph,
        setTemperatureGraph
    };

    return (
        <SlidersContext.Provider value={context}>
            {props.children}
        </SlidersContext.Provider>
    );
}
