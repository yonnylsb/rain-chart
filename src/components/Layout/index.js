import React, { useContext } from 'react';
import Container from '../Container';
import Slider from '../Slider';
import rainchartConstants from '../../common/constants';
import Pressure from '../../components/Pressure';
import { SlidersContext } from '../../providers/slidersContext';
import './Layout.scss';

function Layout(props){
    const {
        setPressureGraph,
        setTemperatureGraph
    } = useContext(SlidersContext);
    return (
        <section className="layout">
            <Container>
                <Slider
                    title="Pressure [hPa]"
                    onSliderChange={setPressureGraph}
                    min={rainchartConstants.pressure.min}
                    max={rainchartConstants.pressure.max}
                />
                <Pressure
                    minX={0}
                    maxX={7}
                    stepX={1}
                    textX="DAYS"
                    minY={0}
                    maxY={100}
                    stepY={20}
                    textY="% CHANCE OF RAIN"
                />
            </Container>
            <Container>
                <Slider
                    title="Temperature [Â°C]"
                    onSliderChange={setTemperatureGraph}
                    min={rainchartConstants.temperature.min}
                    max={rainchartConstants.temperature.max}
                />
                <Pressure
                    minX={0}
                    maxX={7}
                    stepX={1}
                    textX="DAYS"
                    minY={0}
                    maxY={100}
                    stepY={20}
                    textY="% AMOUNT OF RAIN"
                />
            </Container>
        </section>
    );
}

export default Layout;