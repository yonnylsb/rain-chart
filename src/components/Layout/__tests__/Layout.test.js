import React from 'react';
import renderer from 'react-test-renderer';
import { SlidersContext } from '../../../providers/slidersContext';
import Layout from '../index';

test('<Layout/>', () => {
    const mockContext = {
        pressureGraph: 123,
        setPressureGraph: ()=>{},
        temperatureGraph: 12,
        setTemperatureGraph: ()=>{},
        daysValues: [1,2,3,4,5]
    };
    const component = renderer.create(
        <SlidersContext.Provider value={mockContext}>
            <Layout></Layout>
        </SlidersContext.Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
