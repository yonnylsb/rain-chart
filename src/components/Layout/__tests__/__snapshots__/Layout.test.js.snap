// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> 1`] = `
<section
  className="layout"
>
  <div
    className="container"
  >
    <form
      className="slider"
    >
      <label
        className="slider-label"
      >
        Pressure [hPa]
      </label>
      <input
        className="slider-input"
        max={1030}
        min={970}
        onChange={[Function]}
        step="1"
        type="range"
        value={1000}
      />
      <div
        className="slider-value"
      >
        1000
      </div>
    </form>
    <svg
      viewBox="0 0 800 460"
    >
      <path
        className="cor__upper"
        d="M 105.71428571428571 NaN L 211.42857142857142 NaN L 317.1428571428571 NaN L 422.85714285714283 NaN L 528.5714285714286 NaN L 634.2857142857142 NaN L 739.9999999999999 NaN"
      />
      <path
        className="cor__med"
        d="M 105.71428571428571 NaN L 211.42857142857142 NaN L 317.1428571428571 NaN L 422.85714285714283 NaN L 528.5714285714286 NaN L 634.2857142857142 NaN L 739.9999999999999 NaN"
      />
      <path
        className="cor__lower"
        d="M 105.71428571428571 NaN L 211.42857142857142 NaN L 317.1428571428571 NaN L 422.85714285714283 NaN L 528.5714285714286 NaN L 634.2857142857142 NaN L 739.9999999999999 NaN"
      />
      <line
        className="cor__y-axis__line"
        x1="50"
        x2="50"
        y1="0"
        y2="410"
      />
      <line
        className="cor__x-axis__line"
        x1="50"
        x2="780"
        y1="410"
        y2="410"
      />
      <text
        className="cor__y-axis__text"
        x={20}
        y={60}
      >
        % CHANCE OF RAIN
      </text>
      <text
        className="cor__x-axis__text"
        x={400}
        y={450}
      >
        DAYS
      </text>
      <g
        className="cor__y-axis"
      >
        <text
          x={0}
          y={460}
        >
          0
        </text>
        <text
          x={0}
          y={380}
        >
          20
        </text>
        <text
          x={0}
          y={300}
        >
          40
        </text>
        <text
          x={0}
          y={220}
        >
          60
        </text>
        <text
          x={0}
          y={140}
        >
          80
        </text>
        <text
          x={0}
          y={60}
        >
          100
        </text>
      </g>
      <g
        className="cor__x-axis"
      >
        <text
          x={0}
          y={460}
        >
          
        </text>
        <text
          x={105.71428571428571}
          y={460}
        >
          1
        </text>
        <text
          x={211.42857142857142}
          y={460}
        >
          2
        </text>
        <text
          x={317.1428571428571}
          y={460}
        >
          3
        </text>
        <text
          x={422.85714285714283}
          y={460}
        >
          4
        </text>
        <text
          x={528.5714285714286}
          y={460}
        >
          5
        </text>
        <text
          x={634.2857142857142}
          y={460}
        >
          6
        </text>
        <text
          x={739.9999999999999}
          y={460}
        >
          7
        </text>
      </g>
    </svg>
  </div>
  <div
    className="container"
  >
    <form
      className="slider"
    >
      <label
        className="slider-label"
      >
        Temperature [Â°C]
      </label>
      <input
        className="slider-input"
        max={35}
        min={10}
        onChange={[Function]}
        step="1"
        type="range"
        value={23}
      />
      <div
        className="slider-value"
      >
        23
      </div>
    </form>
    <svg
      viewBox="0 0 800 460"
    >
      <rect
        className="aor__bars"
        height={2}
        width="40px"
        x={87.71428571428571}
        y={458}
      />
      <rect
        className="aor__bars"
        height={4}
        width="40px"
        x={193.42857142857142}
        y={456}
      />
      <rect
        className="aor__bars"
        height={6}
        width="40px"
        x={299.1428571428571}
        y={454}
      />
      <rect
        className="aor__bars"
        height={8}
        width="40px"
        x={404.85714285714283}
        y={452}
      />
      <rect
        className="aor__bars"
        height={10}
        width="40px"
        x={510.57142857142856}
        y={450}
      />
      <rect
        className="aor__bars"
        height={NaN}
        width="40px"
        x={616.2857142857142}
        y={NaN}
      />
      <rect
        className="aor__bars"
        height={NaN}
        width="40px"
        x={721.9999999999999}
        y={NaN}
      />
      <line
        className="aor__y-axis__line"
        x1="50"
        x2="50"
        y1="0"
        y2="410"
      />
      <line
        className="aor__x-axis__line"
        x1="50"
        x2="780"
        y1="410"
        y2="410"
      />
      <text
        className="aor__y-axis__text"
        x={20}
        y={60}
      >
        (mm) AMOUNT OF RAIN
      </text>
      <text
        className="aor__x-axis__text"
        x={400}
        y={450}
      >
        DAYS
      </text>
      <g
        className="aor__y-axis"
      >
        <text
          x={0}
          y={460}
        >
          0
        </text>
        <text
          x={0}
          y={420}
        >
          20
        </text>
        <text
          x={0}
          y={380}
        >
          40
        </text>
        <text
          x={0}
          y={340}
        >
          60
        </text>
        <text
          x={0}
          y={300}
        >
          80
        </text>
        <text
          x={0}
          y={260}
        >
          100
        </text>
        <text
          x={0}
          y={220}
        >
          120
        </text>
        <text
          x={0}
          y={180}
        >
          140
        </text>
        <text
          x={0}
          y={140}
        >
          160
        </text>
        <text
          x={0}
          y={100}
        >
          180
        </text>
        <text
          x={0}
          y={60}
        >
          200
        </text>
      </g>
      <g
        className="aor__x-axis"
      >
        <text
          x={0}
          y={460}
        >
          
        </text>
        <text
          x={105.71428571428571}
          y={460}
        >
          1
        </text>
        <text
          x={211.42857142857142}
          y={460}
        >
          2
        </text>
        <text
          x={317.1428571428571}
          y={460}
        >
          3
        </text>
        <text
          x={422.85714285714283}
          y={460}
        >
          4
        </text>
        <text
          x={528.5714285714286}
          y={460}
        >
          5
        </text>
        <text
          x={634.2857142857142}
          y={460}
        >
          6
        </text>
        <text
          x={739.9999999999999}
          y={460}
        >
          7
        </text>
      </g>
    </svg>
  </div>
</section>
`;
